### PR TITLE
ec2_vpc_nacl and ec2_vpc_nacl_info migrate to AnsibleAWSModule and add tests

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Attempt to revive #37794

- Update ec2_vpc_nacl and ec2_vpc_nacl_info to use AnsibleAWSModule
- Update ec2_vpc_nacl and ec2_vpc_nacl_info docs to fix sanity test issues
- Add Integration tests for ec2_vpc_nacl and ec2_vpc_nacl_info
- Add missing AWS permissions

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

test/sanity/ignore.txt
hacking/aws_config/testing_policies/compute-policy.json
hacking/aws_config/testing_policies/network-policy.json
lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py
lib/ansible/modules/cloud/amazon/ec2_vpc_nacl_info.py

##### ADDITIONAL INFORMATION

Mostly test suite work

### CODE REVIEW

File: test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        egress:
+          - [100, 'all', 'allow', '0.0.0.0/0', null, null, null, null]
+        state: 'present'
+      register: nacl
+
+    - assert:
+        that:
+          - nacl.nacl_id
+
+    - name: remove and add an entry
```
Comment (jillr) [inline test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml]: For clarity, did you intend to also add maybe an egress rule? Or should this task just be named `remove an entry`?
Also, what errors are anticipated with the `ignore_errors` on this task?

File: test/integration/targets/ec2_vpc_nacl/tasks/subnet_ids.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  register: nacl
+
+- name: assert the network ACL changed
+  assert:
+    that:
+      - nacl.changed
+      - nacl.nacl_id.startswith('acl-')
+
+- set_fact:
+    nacl_id: "{{ nacl.nacl_id }}"
```
Comment (jillr) [inline test/integration/targets/ec2_vpc_nacl/tasks/subnet_ids.yml]: It does not appear this is being used anywhere.

**Review by jillr — COMMENTED** [review]
Couple small queries, otherwise looks good.

File: test/integration/targets/ec2_vpc_nacl/tasks/subnet_ids.yml (hunk 2 above)
Comment (tremble) [reply test/integration/targets/ec2_vpc_nacl/tasks/subnet_ids.yml]: I'd intended to add a test that we were actually getting the same NACL back and not doing something screwy like deleting and recreating

File: test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml (hunk 1 above)
Comment (tremble) [reply test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml]: See the FIXME comments, IPv6 support is currently missing from ec2_vpc_nacl (but there in ec2_vpc_nacl_info)

 #50004 added support but needs a little work (and looks like it's abandoned)
(edited to fix PR link)

Comment (tremble) [reply test/integration/targets/ec2_vpc_nacl/tasks/ipv6.yml]: (oh and the name's a hangup from @s-hertel's original work)

**Review by jillr — APPROVED** [review]

**Pull request merged**